The 2 failing test:

```
[ERROR] Tests run: 12, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 114.0 s <<< FAILURE! -- in org.apache.hadoop.yarn.server.resourcemanager.TestRMHA
[ERROR] org.apache.hadoop.yarn.server.resourcemanager.TestRMHA.testTransitionedToStandbyShouldNotHang -- Time elapsed: 100.0 s <<< ERROR!
java.util.concurrent.TimeoutException: testTransitionedToStandbyShouldNotHang() timed out after 100000 milliseconds
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	Suppressed: java.lang.InterruptedException
		at java.base/java.lang.Object.wait(Native Method)
		at java.base/java.lang.Thread.join(Thread.java:1313)
		at java.base/java.lang.Thread.join(Thread.java:1381)
		at org.apache.hadoop.yarn.server.resourcemanager.TestRMHA.testTransitionedToStandbyShouldNotHang(TestRMHA.java:538)
		at java.base/java.lang.reflect.Method.invoke(Method.java:569)
		... 2 more

....

[ERROR] org.apache.hadoop.yarn.server.resourcemanager.webapp.TestRMWebServicesReservation.testUpdateReservation(int, boolean)[6] -- Time elapsed: 3.587 s <<< FAILURE!
org.opentest4j.AssertionFailedError: 

Expecting:
 <200>
to be equal to:
 <400>
but was not.
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at org.apache.hadoop.yarn.webapp.WebServicesTestUtils.assertResponseStatusCode(WebServicesTestUtils.java:136)
	at org.apache.hadoop.yarn.server.resourcemanager.webapp.TestRMWebServicesReservation.updateReservationTestHelper(TestRMWebServicesReservation.java:1168)
	at org.apache.hadoop.yarn.server.resourcemanager.webapp.TestRMWebServicesReservation.testUpdateReservation(TestRMWebServicesReservation.java:447)
```	

I can not pass these 2 in the trunk code neither, and seems unrelated to the change.

--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/SecretManager.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/SecretManager.java
@@ -184,9 +184,9 @@ public abstract class SecretManager<T extends TokenIdentifier> {
   }
 
   /**
-   * Validate the secretKey length is equal to the selected config
+   * Validate the secretKey length is equal to the selected config.
    * @param secretKey secretKey
-   * @return true if the length is equal
+   * @return true if the secretKey length is equal to the currently configured length
    */
   protected boolean validateSecretKeyLength(byte[] secretKey) {
     return secretKey.length * 8 == selectedLength;

--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/AMRMTokenSecretManager.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/AMRMTokenSecretManager.java
@@ -345,10 +345,7 @@ public class AMRMTokenSecretManager extends
 
   private AMRMTokenSecretManagerState getTokenState(RMState state) {
     AMRMTokenSecretManagerState result = state.getAMRMTokenSecretManagerState();
-    if (result != null) {
-      result = validateAndUpdateState(result);
-    }
-    return result;
+    return result == null ? null : validateAndUpdateState(result);
   }
 
   private AMRMTokenSecretManagerState validateAndUpdateState(AMRMTokenSecretManagerState state) {

